roles/db-copy/tasks/active-copy.yml

```diff
@@ -0,0 +1,155 @@
+    line: "SID_LIST_{{ listener_name }}=(SID_LIST=(SID_DESC=(GLOBAL_DBNAME={{ oracle_sid }}{{ standby_suffix }}.{{ db_domain }})(ORACLE_HOME={{ oracle_home }})(SID_NAME={{ oracle_sid }})))"
+    owner: "{{ grid_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Reload listener
+  shell: "{{ grid_home }}/bin/lsnrctl reload {{ listener_name }}"
+  environment:
+    ORACLE_HOME: "{{ grid_home }}"
+    PATH: "{{ grid_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ grid_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Create directories
+  file:
+    path: "{{ oracle_base }}/admin/{{ oracle_sid }}/adump"
+    state: directory
+    owner: "{{ oracle_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Create auxiliary init file
+  shell: |
+    set -o pipefail
+    (echo "db_name={{ db_name }}"; echo "db_unique_name={{ db_name }}{{ standby_suffix }}") > {{ oracle_home }}/dbs/init{{ oracle_sid }}.ora
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Start auxiliary instance
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/sqlplus -s / as sysdba << EOF
+    startup nomount pfile={{ oracle_home }}/dbs/init{{ oracle_sid }}.ora force
+    alter system register;
+    host sleep 60
+    EOF
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Set sys password for auxiliary instance
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/orapwd file={{ oracle_home }}/dbs/orapw{{ db_name }} force=y password={{ sys_pass }}
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Duplicate primary database
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/rman << EOF
+    connect target "sys/{{ sys_pass }}@//{{ lookup('env', 'PRIMARY_IP_ADDR') }}:{{ listener_port | default(1521, true) }}/{{ db_name }}{% if db_domain|default('', true)|length > 0 %}.{{ db_domain }}{% endif %}"
+    connect auxiliary "sys/{{ sys_pass }}@//{{ lookup('env', 'INSTANCE_IP_ADDR') }}:{{ listener_port | default(1521, true) }}/{{ db_name }}{{ standby_suffix }}{% if db_domain|default('', true)|length > 0 %}.{{ db_domain }}{% endif %}"
+    duplicate target database for standby from active database
+    password file
+    spfile
+      set "db_unique_name"="{{ db_name }}{{ standby_suffix }}"
+    section size {{ section_size | default('32G', true) }}
+    using compressed backupset;
```

Comment:
If using a compressed backupset, we'll likely be CPU-bound and could probably benefit from multiple channels

Reply:
I will check and modify to replace with parameters for different aspects of duplication

Reply:
for duplication it is using number of channels set at primary:
...
        "Starting backup at 21-JUL-20", 
        "allocated channel: ORA_DISK_1", 
        "channel ORA_DISK_1: SID=401 device type=DISK", 
        "allocated channel: ORA_DISK_2", 
        "channel ORA_DISK_2: SID=31 device type=DISK", 
        "allocated channel: ORA_DISK_3", 
        "channel ORA_DISK_3: SID=139 device type=DISK", 
        "allocated channel: ORA_DISK_4", 
        "channel ORA_DISK_4: SID=276 device type=DISK", 
        "Finished backup at 21-JUL-20", 
        "", 
        "Starting restore at 21-JUL-20", 
        "using channel ORA_AUX_DISK_1", 
        "using channel ORA_AUX_DISK_2", 
        "using channel ORA_AUX_DISK_3", 
        "using channel ORA_AUX_DISK_4", 
...


I also added variables for section size and compression of backupsets

Reply:
put it as template and added task to get template copied to remote node before doing duplication with cmdfile=...